### PR TITLE
Ec git guide

### PR DESCRIPTION
**Pull request by ericthewry**
Heres a bunch of General Git info.  Let me know if you have any feedback @rwoll

### CODE REVIEW

File: git-guide.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+control software is that it allows for divergent copies to be checked out. This
+allows for managing multiple versions of code simultaneously. You can then merge
+branches together to combine the changes that have been made. Accepted best
+practices suggest having a `master` branch that contains the best and most
+functional version of the code. Then for each addition or change, create a new
+branch. Once that change has been finalized, merge it back into the current branch.
+Typically, if `branchB` is checked out from `branchA`, `branchB` will eventually
+be merged back into `branchA`.
+
+Some handy branching commands:
```
Comment (rwoll) [inline git-guide.md]: Add a line break after here so the table renders properly.

File: git-guide.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+| Action                         | Description                                                |
+|:------------------------------:|:----------------------------------------------------------:|
+| `git checkout existing-branch` | *Checks out the branch `existing-branch`*                  |
+| `git checkout -b new-branch`   | *Creates a new branch `new-branch` and checks it out*      |
+| `git merge branchA`            | *merges branch existing `branchA` into the current branch* |
+| `git branch -a`                | *lists all branches, local and remote*                     |
+
+
+## Related links
```
Comment (rwoll) [inline git-guide.md]: Missing links! Other than that, really nice guid. Depending on how we use it in the course materials, we may wan't to simplify it and explain things in the context of the ALIASes we created?

**Comment by rwoll** [comment]
I'm going to close this, but leave the branch alive so we can move the guides.

**Comment by ericthewry** [comment]
Just a heads up.. Generally it is anti-pattern to close pull requests unless you are also going to delete the branch, i.e. If you aren't going to delete the branch, leave it open. It just results in a lot more work and clutter for only a little gain.

That aside, I'm confused about what you mean by 'move the guides'.  Where do you want to move them?

File: git-guide.md (hunk 2 above)
Comment (ericthewry) [inline git-guide.md]: I'm not sure.  My goal here was to give a general backgrounder on Git basics.  Most people wont read it.  They will read their specific guide, though, and that's where we'll explain the basics in terms of our aliases.  A note about how we've simplified it for this course might be useful if we pursue this pattern.

At least that's what I was thinking.  What was your architectural thought?

**Comment by rwoll** [comment]
Thanks for the heads up--sorry for my lack of explanation. Here's what I had in mind:

By 'move the guides', I really meant I figured we'd (interactively) rebase each of the `ec-*` branches with the `master` so you could get my commits with the new site layout (including the `rm`s pertaining to the old site files), and  the added Travis CI files. Once that's accomplished, the `ec-*` branches would have the guide files you created and the updated site related files. We could make our edits and then merge into master in a new PR--although if you just want to reopen this PR that might make more sense.

@ericthewry Let me now if that seems ok or if I'm overcomplicating it.

File: git-guide.md (hunk 2 above)
Comment (rwoll) [inline git-guide.md]: I like that :+1: I think it's definitely good to have a "how-to" guide and then a more general background guide so people get a sense of what's happening behind the scenes.

**Comment by rwoll** [comment]
@ericthewry On second thought, for what we are doing I'm overcomplicating it -- I think it would be fine to have the extra commits left from doing the following operation (although squashing some before the final merge into master would probably be appropriate):

```bash
git checkout ec-git-guide
git merge master
git push origin ec-git-guide
...work on guides and have CI tests run...
```

**Comment by rwoll** [comment]
@ericthewry looks good. We'll have to edit the CSS to get the tables to render appropriately, but I plan to do that in a new branch after everything is together and we can see what we want to change.

**Comment by ericthewry** [comment]
@rwoll let me know if I set up jekyll right and merge.. I'm too tired to set up gems right now.

**Comment by rwoll** [comment]
@ericthewry Everything passed and a local run looked good! :+1:

**Pull request merged**